experimental/segmenter/src/lstm_bies.rs

```diff
@@ -120,22 +120,22 @@ impl<'l> Lstm<'l> {
         warr: ArrayBase<ViewRepr<&f32>, Dim<[usize; 2]>>,
         uarr: ArrayBase<ViewRepr<&f32>, Dim<[usize; 2]>>,
         barr: ArrayBase<ViewRepr<&f32>, Dim<[usize; 1]>>,
-    ) -> (Array1<f32>, Array1<f32>) {
+    ) -> Result<(Array1<f32>, Array1<f32>), Error> {
         // i, f, and o respectively stand for input, forget, and output gates
         let s_t = x_t.dot(&warr) + h_tm1.dot(&uarr) + barr;
-        let hunits = uarr.shape()[0];
+        let hunits = *uarr.shape().get(0).ok_or(Error::DimensionMismatch)?;
         let i = math_helper::sigmoid_arr1(s_t.slice(ndarray::s![..hunits]));
         let f = math_helper::sigmoid_arr1(s_t.slice(ndarray::s![hunits..2 * hunits]));
         let _c = math_helper::tanh_arr1(s_t.slice(ndarray::s![2 * hunits..3 * hunits]));
         let o = math_helper::sigmoid_arr1(s_t.slice(ndarray::s![3 * hunits..]));
         let c_t = i * _c + f * c_tm1;
         let h_t = o * math_helper::tanh_arr1(c_t.view());
-        (h_t, c_t)
+        Ok((h_t, c_t))
     }
 
     /// `word_segmenter` is a function that gets a "clean" unsegmented string as its input and returns a BIES (B: Beginning, I: Inside, E: End,
     /// S: Single) sequence for grapheme clusters. The boundaries of words can be found easily using this BIES sequence.
-    pub fn word_segmenter(&self, input: &str) -> String {
+    pub fn word_segmenter(&self, input: &str) -> Result<String, Error> {
```

Comment:
To fix `clippy::indexing_slicing`, we now propagate errors from `word_segmenter`. 

Question: is it sufficient to detect the LSTM model broken in `Lstm::try_new()`? It seems strange to me that we now have to handle error from `word_segmenter`, which leads to error handling at callsites like `LstmSegmenter::segment_str()`. The testcases in `lstm.rs` has example that we now have to write

```
.segment_str(TEST_STR)
.expect("LSTM payload is broken")`
```

Ideally, like line or word segmenter, we should be able to detect payload data error when we load it from the data provider in a segmenter's constructor so that creating iterators generates no errors.

Reply:
We don't use error states for corrupted payloads. If the payload is broken, just return some garbage value (garbage-in-garbage-out). The important thing is that we don't panic.

Reply:
Since as long as external crate (rust-ndarray) has no bug, `word_segmenter` won't return error now. Should I use `reachable!` instead of `Result<...>`?

Reply:
As long as external crate (rust-ndarray) has no bug, word_segmenter won't return error now. Should I use reachable! instead of Result<...>?

Reply:
It looks like this code is reachable from the `segment_str` function, which seems like a reasonable thing to make fallible. If the function is already fallible, then it's easy to bubble up more types of errors. Then again, segment_str is not currently fallible, and it's weird if this is the only type of error that can occur.

Reply:
> As long as external crate (rust-ndarray) has no bug

`rust-ndarray` actually has quite a few documented panics. Are we 100% sure that we meet all the preconditions and it won't panic?

Reply:
Conclusion from segmenter meeting with @sffc @makotokato @aethanyc 

If the code is 100% unreachable, then panic 
If the code is not 100% unreachable, but the function is fallible, return an error 
If the function is not fallible, but the code is not 100% unreachable, then use a garbage value